src/Esfa.Vacancy.Register.Api/Orchestrators/GetApprenticeshipVacancyOrchestrator.cs

```diff
@@ -32,30 +33,27 @@ public class GetApprenticeshipVacancyOrchestrator
 
         public async Task<ApprenticeshipVacancy> GetApprenticeshipVacancyDetailsAsync(string id)
         {
-            ApprenticeshipVacancy vacancy = null;
-            int parsedId;
-            if (!int.TryParse(id, out parsedId))
+            ApprenticeshipVacancy apprenticeshipVacancy;
+            if (!int.TryParse(id, out var parsedId))
             {
                 throw _validationExceptionBuilder.Build(
                     ErrorCodes.GetApprenticeship.VacancyReferenceNumberNotInt32,
                     ErrorMessages.GetApprenticeship.VacancyReferenceNumberNotNumeric);
             }
-
             if (VacancyVersionHelper.IsRaaVacancy(parsedId))
             {
                 var response = await _mediator.Send(new GetApprenticeshipVacancyRequest() { Reference = parsedId })
                     .ConfigureAwait(false);
-                vacancy = _mapper.MapToApprenticeshipVacancy(response.ApprenticeshipVacancy);
+                apprenticeshipVacancy = _mapper.MapToApprenticeshipVacancy(response.ApprenticeshipVacancy);
             }
             else
             {
-                var liveVacancy = _recruitClient.GetVacancy(parsedId);
-                if (liveVacancy == null) throw new ResourceNotFoundException(Domain.Constants.ErrorMessages.VacancyNotFoundErrorMessage);
-
-                vacancy = await _recruitMapper.MapFromRecruitVacancy(liveVacancy).ConfigureAwait(false);
+                var vacancy = _recruitClient.GetVacancy(parsedId);
+                if (vacancy == null || vacancy.ViewType.Equals(ClosedVacancyDocumentType))
```

Comment:
This should really be in the client package (https://github.com/SkillsFundingAgency/das-shared-packages/blob/master/SFA.DAS.Recruit.Vacancies.Client/SFA.DAS.Recruit.Vacancies.Client/Client.cs). So there should be a GetLiveVacancy method which excludes those with a ClosedVacancy view type. We don't want the Vacancy API to know about view types. The client package already knows about them.

Reply:
I agree with this. Also the GetVacancy endpoint in the Recruit client is not named correctly... it should be named GetPublishedVacancy.